langchain/callbacks/streaming_last_response_callback.py

```diff
@@ -0,0 +1,449 @@
+            import tiktoken
+            enc = tiktoken.get_encoding("cl100k_base")
+
+            stream = StreamingLastResponseCallbackHandler.from_agent_type(agent=AgentType.ZERO_SHOT_REACT_DESCRIPTION)
+
+            @stream.postprocess(sliding_window_step=1, window_size=3)
+            def postprocess_func(tokens: List[str]) -> List[str]:
+                sentence = "".join(tokens).replace("Python", "LangChain")
+                words = [enc.decode([t]) for t in enc.encode(sentence)]  # postprocess output can have different size!
+                return words
+
+            def _run():
+                agent.run("Is python good?", callbacks=[stream])
+            threading.Thread(target=_run).start()
+
+            for token in stream:
+                print(token, end="", flush=True)
+    """
+
+    def __init__(
+        self,
+        answer_prefix_phrases: List[str] = ["Final Answer:"],
+        error_stop_streaming_phrases: List[str] = [],
+        case_sensitive_matching: bool = False,
+        output_stream_prefix: bool = False,
+        tiktoken_encoding: str = "cl100k_base",
+    ) -> None:
+        """
+        Args:
+            answer_prefix_phrases: List of phrases that indicate that the next
+                token is the final answer. Multiple phrases are allowed, the first
+                one that matches will be used. Phrase matching is case sensitive.
+                Some phrases can be a substring of the other phrase. If multiple
+                phrases are detected, the longest one will be used.
+                Example: ["Final Answer:", "Final Answer"]
+            error_stop_streaming_phrases: List of phrases that indicate that the
+                next token is an error message and that the streaming should stop.
+                Multiple phrases are allowed, the first one that matches will stop
+                the streaming. Phrase matching is case sensitive.
+            case_sensitive_matching: If True, the answer_prefix_phrases and
+                error_stop_streaming_phrases will be matched case sensitive.
+                Default: False
+            output_stream_prefix: If True, the output stream will include the
+                found answer_prefix_phrases. If False, the output stream will
+                only include the final answer and exclude the matched
+                answer_prefix_phrases. Default: False
+            tiktoken_encoding: The encoding to use for the tiktoken.
+                Default: "cl100k_base"
+        """
+        super().__init__()
+
+        try:
+            import tiktoken
+        except ImportError:
+            raise ImportError(
+                "Could not import tiktoken python package. "
+                "This is needed in order to calculate detection_windows_size for StreamingLastResponseCallbackHandler"
+                "Please install it with `pip install tiktoken`."
+            )
+        self._enc = tiktoken.get_encoding(tiktoken_encoding)
```

Comment:
is it worth letting folks pass in whatever tokenize they want (can still default to tiktoken "cl100k_base")?

Reply:
> is it worth letting folks pass in whatever tokenize they want (can still default to tiktoken "cl100k_base")?

@baskaryan Thank you for the suggestion, I have implemented it, please check it!

https://github.com/thaiminhpv/langchain/blob/4c8bdceecf6a8eef06968dadea06878bc4875676/libs/langchain/langchain/callbacks/streaming_last_response_callback.py#L106-L135